### DISCUSSION TITLE
CancellationToken的token.register

### DISCUSSION THREAD

**Discussion by Awweii** [BYJRK/DotNet-Discussions#386] [C# 技术交流]
.NET6的控制台项目，学习CancellationTokenSource的，有如下疑问
![problem](https://github.com/user-attachments/assets/1a14f047-520d-41a6-858c-e7e2f2dee8d6)

**Comment by WHYBBE** [comment]
这个问题很有意思，我试了一下确实在.net6.0是不会**立刻**输出回调的结果的，7.0、8.0、9.0都是能先于catch内的内容输出

我想大概是回调的时机应该是不确定的，不应该太依赖于这个时间（或者有其他机制能保证这个执行顺序？）

还有就是为什么不是立刻，因为被`Console.ReadKey`给阻塞了
